### PR TITLE
User profile functionality - backend

### PR DESCRIPTION
**Pull request by munteanumarco**

### CODE REVIEW

File: EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs [hunk 1]
```diff
@@ -1,14 +1,17 @@
 ﻿using BusinessLayer.DTOs;
+using DataAccessLayer.Models;
 using Microsoft.AspNetCore.Identity;
 
 namespace BusinessLayer.Interfaces
 {
     public interface IUserService
     {
-		Task<IdentityResult> CreateUserAsyncLogic(UserDto user);
+		Task<IdentityResult> CreateUserAsync(RegisterUserDto user);
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs]: padding

File: EventPlannerAPI/EventPlannerAPI/Properties/launchSettings.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "EventPlannerAPI": {
       "commandName": "Project",
       "dotnetRunMessages": true,
-      "launchBrowser": true,
+      "launchBrowser": false,
```
Comment (HellHound97) [inline EventPlannerAPI/EventPlannerAPI/Properties/launchSettings.json]: Why?

File: EventPlannerAPI/EventPlannerAPI/Controllers/UserController.cs [hunk 3]
```diff
@@ -62,5 +63,63 @@ public async Task<IActionResult> SetNewPassword([FromBody] SetNewPasswordDto set
...
+            {
+                return Ok(await _userService.UpdateUserProfileDetailsAsync(userId, userDetails));
+            }
+            catch (EventPlannerException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+            catch (Exception)
+            {
+                return Problem("Something went wrong.");
```
Comment (HellHound97) [inline EventPlannerAPI/EventPlannerAPI/Controllers/UserController.cs:128]: Why do you use Problem and not use BadRequest?

Comment (munteanumarco) [reply EventPlannerAPI/EventPlannerAPI/Controllers/UserController.cs:128]: cause in that case is not a bad request (4xx status codes means user did something wrong)
in our case is a server error and we need a 5xx status code

**Review by HellHound97 — APPROVED** [review]

**Review by gheorghe-sava — APPROVED** [review]

**Pull request merged**